### ISSUE TITLE
uid hard coded in v4.0.1 ldap_auth.py

### ISSUE THREAD

**Issue by rbur004** [mytardis/mytardis#1844]
in v4.0.1 : A hardcoded 'uid' has snuck back into tardis/tardis_portal/ldap_auth.py code in line 165

Diff with the copy you asked me to test.
-            if ldap_result[0][1]['uid'][0] == username:
+            if ldap_result[0][1][self._login_attr][0] == username.encode():

It looks to be correct in the v4.1 code base, which is probably the one I tested against.

**Comment by jameswettenhall** [comment]
Ah, thanks @rbur004, we had to cherry-pick these changes from the series-4.1 / develop branches and it looks like we missed this one.  Apologies.  I'll try to get v4.0.2 tagged and released soon including a fix for this.  From memory, I gave you a (more correctly) cherry-picked version of this file to patch on top of MyTardis v4.0.  I don't remember you testing a v4.1 pre-release branch but I could be wrong.
